### PR TITLE
Malformed CAR panics and excessive memory usage

### PR DESCRIPTION
**Pull request by bhaskarvilles**
## Description

Decoding CAR data from untrusted user input can cause:

    Panics:
        - Out of bound memory access
        - Out of memory
        - Divide by zero

## Excessive memory usage

Such panics can be triggered by intentionally malformed CARv1 data, including CARv1 data within a CARv2 container; and also CARv2 data with excessively large indexes.

These vulnerabilities are not known to be exploited in the wild and were discovered primarily with the use of code fuzzing tooling.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -33,7 +33,7 @@ require (
 	github.com/ipfs/go-ipld-cbor v0.0.6
 	github.com/ipfs/go-log/v2 v2.5.1
 	github.com/ipfs/go-metrics-prometheus v0.0.2
-	github.com/ipld/go-car v0.3.3
+	github.com/ipld/go-car v2.3.0
```
Comment (frrist) [inline go.mod:36]: Since this is v2 the path must be: `github.com/ipld/go-car/v2`

**Review by frrist — CHANGES_REQUESTED** [review]
Thanks, @bhaskarvilles, left some small feedback.

File: go.mod (hunk 1 above)
Comment (frrist) [reply go.mod:36]: @bhaskarvilles please tag me for a second review once this has been addressed. Currently, it is still unresolved.

Comment (bhaskarvilles) [reply go.mod]: @frrist Resolved, now you can merge changes.

**Comment by bhaskarvilles** [comment]
> Thanks, @bhaskarvilles, left some small feedback. 

Done ! Resolved.

**Comment by frrist** [comment]
@bhaskarvilles, unfortunat this doesn't build due to other dependencies needing updates.

**Comment by bhaskarvilles** [comment]
> @bhaskarvilles, unfortunat this doesn't build due to other dependencies needing updates. You can try this yourself by running `make build`

I have to compile myself in my local system or what ??

**Comment by bhaskarvilles** [comment]
> @bhaskarvilles, unfortunat this doesn't build due to other dependencies needing updates. You can try this yourself by running `make build`

Update??

**Comment by frrist** [comment]
Because PR's from forks do not run CI we cannot tell if this code builds. I tried you build your branch locally and it failed because there are other changes needed to update this dependency. So there are two options:
1. Build this project yourself (on your machine) and fix the failures
2. Wait and we will incorporate this change at a later date.

**Comment by bhaskarvilles** [comment]
> Because PR's from forks do not run CI we cannot tell if this code builds. I tried you build your branch locally and it failed because there are other changes needed to update this dependency. So there are two options:
> 1. Build this project yourself (on your machine) and fix the failures
> 2. Wait and we will incorporate this change at a later date.

I've tried building the changes in my local system also, but it failed. Just drop me a message if you implement these changes in some later dates.

**Comment by frrist** [comment]
okay will reopen when we reach that point.

**Pull request closed without merging**